src/rowbinary/de.rs

```diff
@@ -242,42 +400,204 @@ impl<'data> Deserializer<'data> for &mut RowBinaryDeserializer<'_, 'data> {
             }
         }
 
-        visitor.visit_seq(Access {
-            deserializer: self,
+        let inner_data_type_validator = self.validator.validate(
+            &SerdeType::Tuple,
+            &[
+                DataTypeHint::Tuple,
+                DataTypeHint::Array,
+                DataTypeHint::FixedString,
+                // FIXME: uncomment when there is a way to implement ReverseSeqAccess
```

Comment:
If we somehow implement conditional ReverseSeqAccess (perhaps via the InnerDataTypeValidator, a new variant like ReversedArray), then we can get rid of custom serde for IPv4.